src/proxy/grpc/service.rs

```diff
@@ -111,3 +111,24 @@ pub mod res_body_as_payload {
         }
     }
 }
+
+pub mod unauthenticated {
+    use api::tap as api;
+    use futures::{future, stream};
+    use tower_grpc::{Code, Request, Response, Status};
+
+    #[derive(Clone)]
+    pub struct Unauthenticated;
+
+    impl api::server::Tap for Unauthenticated {
```

Comment:
When the client is not authenticated or authorized, is it better to accept the connection and serve back Unauthenticated errors like this?  Or simply reject the connection?

Reply:
So we could just reject the connection, and it would remove a majority of the changes this PR introduces. However, mainly in testing and especially for debugging tap, I found it useful to be able to see that a client is unauthenticated from tapping a certain pod. I would prefer to keep this, but I'm open to reason on why it should be removed.